src/test/kotlin/com/cultureamp/eventsourcing/example/SkillsCustomisationDraftAggregate.kt

```diff
@@ -0,0 +1,197 @@
+import com.cultureamp.eventsourcing.UpdateEvent
+import com.cultureamp.eventsourcing.example.DraftStatus.Discarded
+import com.cultureamp.eventsourcing.example.DraftStatus.Open
+import com.cultureamp.eventsourcing.example.DraftStatus.Published
+import com.cultureamp.eventsourcing.map
+import com.cultureamp.eventsourcing.sample.StandardEventMetadata
+import org.joda.time.DateTime
+import java.util.UUID
+
+sealed interface SkillsCustomizationDraftCreationCommand : CreationCommand
+sealed interface SkillsCustomizationDraftUpdateCommand : UpdateCommand
+
+data class DraftAddSkill(override val aggregateId: UUID, val skillId: UUID, val label: String?, val description: String?) : SkillsCustomizationDraftCreationCommand, SkillsCustomizationDraftUpdateCommand
+data class DraftRemoveSkill(override val aggregateId: UUID, val skillId: UUID) : SkillsCustomizationDraftCreationCommand, SkillsCustomizationDraftUpdateCommand
+
+data class DraftPublish(override val aggregateId: UUID) : SkillsCustomizationDraftUpdateCommand
+data class DraftDiscard(override val aggregateId: UUID) : SkillsCustomizationDraftUpdateCommand
+
+sealed interface SkillsCustomizationDraftEvent : DomainEvent
+sealed interface SkillsCustomizationDraftCreationEvent : SkillsCustomizationDraftEvent, CreationEvent {
+    val accountId: UUID
+}
+
+sealed interface SkillsCustomizationDraftUpdateEvent : SkillsCustomizationDraftEvent, UpdateEvent
+
+data class DraftSnapshottedCanonicalSkills(
+    val coreSkills: List<CoreSkillSnapshot>,
+    val coreLeadershipSkills: List<CoreLeadershipSkillSnapshot>,
+    val coreSkillGroupId: UUID,
+    val coreLeadershipSkillGroupId: UUID,
+    override val accountId: UUID
+) : SkillsCustomizationDraftCreationEvent
+
+data class DraftSkillAdded(val skillId: UUID, val label: String?, val description: String?, override val accountId: UUID, val addedAt: DateTime) : SkillsCustomizationDraftCreationEvent, SkillsCustomizationDraftUpdateEvent
+data class DraftSkillRemoved(val skillId: UUID, override val accountId: UUID, val removedAt: DateTime) : SkillsCustomizationDraftCreationEvent, SkillsCustomizationDraftUpdateEvent
+
+data class DraftDiscarded(val discardedAt: DateTime) : SkillsCustomizationDraftUpdateEvent
+data class DraftPublished(val publishedAt: DateTime) : SkillsCustomizationDraftUpdateEvent
+
+data class CoreSkillSnapshot(val coreSkill: CoreSkill, val label: String, val description: String, val skillId: UUID)
+data class CoreLeadershipSkillSnapshot(val coreLeadershipSkill: CoreLeadershipSkill, val label: String, val description: String, val skillId: UUID)
+
+interface PublishedSkillsProjection {
+    fun publishedSkillsFor(accountId: UUID): Map<UUID, Skill>
+}
+
+interface AdminProjection {
+    fun isAdminOfAccount(userId: UUID, accountId: UUID): Boolean
+}
+
+object AlwaysAdminAdminProjection : AdminProjection {
+    override fun isAdminOfAccount(userId: UUID, accountId: UUID) = true
+}
+
+data class SkillsCustomisationDraftAggregate(
```

Comment:
I can't believe it, you implemented this aggregate here already. 💯 👍

Reply:
I needs to be extended for skill edits, and groups / assignments but the basic shape is all there 🥳